subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/DefaultSslContextFactory.java

```diff
@@ -100,161 +82,21 @@ public SSLContext load(Map<String, String> props) {
              * https://github.com/gradle/gradle/issues/7842
              * https://github.com/gradle/gradle/issues/2588
              */
-            return SystemProperties.getInstance().withSystemProperties(props, new Factory<SSLContext>() {
-                @Override
-                public SSLContext create() {
-                    return delegate.load(props);
-                }
-            });
+            return SystemProperties.getInstance().withSystemProperties(props, () -> SslContextLoader.load(props));
         }
     }
 
-    private static class SslContextCacheLoader extends CacheLoader<Map<String, String>, SSLContext> {
-        @Override
-        public SSLContext load(Map<String, String> props) {
-            // TODO: We should see if we can go back to using HttpClient again.
-            // This implementation is borrowed from the Apache HttpClient project
-            // https://github.com/apache/httpclient/blob/4.2.2/httpclient/src/main/java/org/apache/http/conn/ssl/SSLSocketFactory.java#L246-L354
-            try {
-                TrustManagerFactory tmFactory = initTrustManagerFactory(props);
-                KeyManagerFactory kmFactory = initKeyManagerFactory(props);
-
-                SSLContext sslcontext = SSLContext.getInstance("TLS");
-                sslcontext.init(kmFactory.getKeyManagers(), tmFactory.getTrustManagers(), null);
+    @NonNullApi
+    private static class SslContextLoader {
+        private static final Logger LOGGER = LoggerFactory.getLogger(SslContextLoader.class);
 
-                return sslcontext;
-            } catch (GeneralSecurityException | IOException e) {
+        public static SSLContext load(Map<String, String> props) {
+            try {
+                return SystemDefaultSSLContextFactory.create();
+            } catch (Exception e) {
+                LOGGER.error("Could not initialize SSL context. Used properties: {}", props);
                 throw new SSLInitializationException(e.getMessage(), e);
```

Comment:
💭 If properties are useful to know in case of failure, should they be part of the exception message as well?

Reply:
There are cases when the exception message is not propagated (e.g. in plugin resolution). Also, I think it might be a bit log for the exception itself.